### PR TITLE
Add missing check for nil expression

### PR DESCRIPTION
**Pull request by fabxc**
This is the cause of #940.
Bit of error message flexibility given up for safety.

@msiebuhr

### CODE REVIEW

File: promql/parse.go [hunk 1]
```diff
@@ -665,6 +660,9 @@ func (p *parser) primaryExpr() Expr {
 	case t.typ.isAggregator():
 		p.backup()
 		return p.aggrExpr()
+
+	default:
+		p.errorf("no valid expression found")
 	}
 	return nil
```
Comment (juliusv) [inline promql/parse.go]: You could remove this line now since it's dead code (legal in Go if there's a default in the switch).

Comment (fabxc) [inline promql/parse.go]: Only if I move `return nil` into the default. Which is also never executed as `p.errorf` panics.
Both options are equally noops.

Comment (juliusv) [inline promql/parse.go]: Ah yeah. Duh.

**Comment by juliusv** [comment]
:+1:

**Pull request merged**